src/Faithlife.Data/DbParameters.cs

```diff
@@ -77,22 +77,64 @@ public static DbParameters Create(Func<int, string> name, IEnumerable<object?> v
 		/// <summary>
 		/// Creates a list of parameters from the properties of a DTO.
 		/// </summary>
-		public static DbParameters FromDto(object dto) =>
-			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => (x.Name, x.GetValue(dto))));
+		public static DbParameters FromDto(object dto, string? suffix = null) =>
+			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => (x.Name + (suffix == null ? "" : $"_{suffix}"), x.GetValue(dto))));
 
 		/// <summary>
 		/// Creates a list of parameters from the properties of a DTO.
 		/// </summary>
-		public static DbParameters FromDto(string name, object dto) =>
-			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => ($"{name}_{x.Name}", x.GetValue(dto))));
+		public static DbParameters FromDto(string name, object dto, string? suffix = null) =>
+			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => ($"{name}_{x.Name}" + (suffix == null ? "" : $"_{suffix}"), x.GetValue(dto))));
 
 		/// <summary>
 		/// Creates a list of parameters from the properties of a DTO.
 		/// <param name="name">A function taking the name of a DTO property as an argument and returning the name of its database parameter.</param>
 		/// <param name="dto">The DTO to retrieve parameters from.</param>
+		/// <param name="suffix">An optional suffix to add to the end of each parameter name after adding an underscore.</param>
 		/// </summary>
-		public static DbParameters FromDto(Func<string, string> name, object dto) =>
-			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => (name(x.Name), x.GetValue(dto))));
+		public static DbParameters FromDto(Func<string, string> name, object dto, string? suffix = null) =>
+			new DbParameters(DtoInfo.GetInfo((dto ?? throw new ArgumentNullException(nameof(dto))).GetType()).Properties.Select(x => (name(x.Name) + (suffix == null ? "" : $"_{suffix}"), x.GetValue(dto))));
+
+		/// <summary>
+		/// Creates a list of parameters from the collective properties of a sequence of DTOs.
+		/// </summary>
+		public static DbParameters FromDtos(IEnumerable<object> dtos)
```

Comment:
Per my other comment (and despite my proposed signature in the issue), `IEnumerable` would be better here so that collections of value types can be used as well.